aepp/exportDatasetToCloudStorage.py

```diff
@@ -0,0 +1,179 @@
+                target_connection_id
+            ],
+            "transformations": [],
+            "scheduleParams": schedule_params
+        }
+        flow_res = self.flow_conn.createFlow(
+            obj = flow_obj,
+            flow_spec_id = flow_spec_id
+        )
+        try:
+            dataflow_id = flow_res["id"]
+            print("dataflowId " + dataflow_id + " has been created")
+        except KeyError:
+            raise RuntimeError(f"Error when creating data flow: {flow_res}")
+
+        #save dataflow_id to config
+        self.utils.save_field_in_config("Platform", "dataflow_id", dataflow_id)
+        return dataflow_id
+
+    def createFlowRun(self, dataflow_id, dataset_id, on_schedule):
+        if not on_schedule:
+            res = self.retryOnNotReadyException(dataflow_id = dataflow_id, dataset_id= dataset_id)
+            flowRunId = res["destinations"][0]["datasets"][0]["statusURL"].rsplit('/', 1)[-1]
+            #check if flowRun has finished
+            finished = False
+            while not finished:
+                try:
+                    run = self.flow_conn.getRun(runId = flowRunId)["items"][0]
+                    run_id = run["id"]
+                    run_started_at = run["metrics"]["durationSummary"]["startedAtUTC"]
+                    run_ended_at = run["metrics"]["durationSummary"]["completedAtUTC"]
+                    run_duration_secs = (run_ended_at - run_started_at) / 1000
+                    run_size_mb = run["metrics"]["sizeSummary"]["outputBytes"] / 1024. / 1024.
+                    run_num_rows = run["metrics"]["recordSummary"]["outputRecordCount"]
+                    run_num_files = run["metrics"]["fileSummary"]["outputFileCount"]
+                    print(f"Run ID {run_id} completed with: duration={run_duration_secs} secs; size={run_size_mb} MB; num_rows={run_num_rows}; num_files={run_num_files}")
+                    finished = True
+                except Exception as e:
+                    print(f"No runs completed yet for flow {self.dataflow_Id}")
+                    time.sleep(30)
+            #push the startDate to a year from now
+            startTime = int(time.time() + 60*60*24*365)
+            op = [
+                {
+                    "op" : "Replace",
+                    "path":"/scheduleParams",
+                    "value":{
+                        "startTime": startTime
+                    }
+
+                }
+            ]
+            flowRes = self.flow_conn.getFlow(dataflow_id)
+            eTag = flowRes["etag"]
+            self.flow_conn.updateFlow(dataflow_id,eTag,op)
+            #disable flow after the adhoc run is completed
+            self.flow_conn.postFlowAction(dataflow_id, "disable")
+
+    def retryOnNotReadyException(self, dataflow_id, dataset_id, max_attempts=3, delay=60):
+        #set up initial delay due to hollow refresh time interval
+        time.sleep(300)
+        attempts = 1
+        while attempts <= max_attempts:
```

Comment:
Retry logic I don't think you need to do that manually, there's a lot of great libraries to handle that in a more streamlined way. For example I've used `tenacity` library before in Python and it is very flexible and works with annotations to achieve this without the need for custom code.

Reply:
This is great feedback, I have been utilizing the new library now.